modules/tests/src/test/scala/dev/profunktor/redis4cats/pubsub/internals/SubscriberSuite.scala

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2018-2021 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats
+package pubsub
+package internals
+
+import cats.effect.IO
+import dev.profunktor.redis4cats.data.RedisChannel
+import scala.concurrent.duration._
+
+class SubscriberSuite extends IOSuite {
+
+  private val waitOnFiber: IO[Unit] = IO.sleep(200.millis)
+
+  private val channel1 = RedisChannel("a")
+  private val channel2 = RedisChannel("b")
+
+  test("subscribe and unsubscribe") {
+    for {
+      subRef <- IO.ref(0)
+      unsubRef <- IO.ref(0)
+      map <- subscriptionMap(subRef.update(_ + 1), unsubRef.update(_ + 1))
+      subscription <- map.subscribe(channel1).compile.toList.start
+      _ <- waitOnFiber
```

Comment:
Instead of sleep which makes for long tests and race conditions in CI, how about [cats effect's testcontrol](https://typelevel.org/cats-effect/docs/core/test-runtime)? Otherwise, if we want to keep the exact runtime:
```suggestion
     waitForSubscribe <- Deferred[IO, Unit]
      subscription <- map.subscribe(channel1).compile.toList.flatTap(_ => waitForSubscribe.complete(())).start
      _ <- waitForSubscribe.get()
```

Reply:
I used the `sleep` initially because that is what `RedisPubSubSpec` uses, but unlike those tests we aren't actually communicating with redis here.

The complicating factor is that `subscribe` is doing quite a lot: it is subscribing, but also returning the messages. In these tests we want to be subscribed (both to redis and to the `Topic`) before we use any of the other operations where we assume there is a subscription.

I am working on a change to the internal `subscribe` signature to make it easier to test (separating the subscription from the message consumption effect), but I still need to clean up these tests a bit more before I'll push that work. Thanks for this comment, I should have given this some more thought before.

On a side note, your `Deferred` suggestion unfortunately doesn't work. The `subscribe` method never terminates on it is own. We need something to call `unsubscribe` or end the stream early.